lib/util/src/vlb.c

```diff
@@ -156,4 +164,6 @@ vlb_fini(void)
             munmap(mem, VLB_ALLOCSZ_MAX);
         }
     }
+
+    vlb_initialized = false;
```

Comment:
FWIW, this just worked (by accident) irrespective of your changes because all spin-lock-init does is set the lock word to zero, which it already is since it resides in the .bss.  Of course, that is implementation dependent (we're using ptrheads, but we should probably just replace it with cmpxchg, ala ~gbecker/spinlock.h).

Reply:
Happy to review a small PR that adds that functionality.

Reply:
I haven't pulled the trigger yet.  The good thing about my implementation is that the cost of the spinlock shows up in the function using it, vs aggregated behind pthread_spin_{lock,unlock) functions that make it difficult to know who's calling them...   Not to mention the zero-overhead cost of my implementation!

Reply:
Zero is my favorite number!